test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/CpsPackageReferenceProjectTests.cs

```diff
@@ -2316,6 +2316,703 @@ public async Task GetInstalledPackages_WithNominationUpdate_ReloadsCache()
+
+                    // This code added because nuGetPackageManager.InstallPackageAsync doesn't do updating ProjectSystemCache
+                    var installed = new LibraryDependency
+                    {
+                        LibraryRange = new LibraryRange(
+                            name: packageB.Id,
+                            versionRange: new VersionRange(packageB.Version),
+                            typeConstraint: LibraryDependencyTarget.Package),
+                    };
+
+                    PackageSpec packageSpec = packageSpecs[i];
+                    packageSpec.TargetFrameworks.First().Dependencies.Add(installed);
+                    DependencyGraphSpec projectRestoreInfo = ProjectJsonTestHelpers.GetDGSpecFromPackageSpecs(packageSpec);
+                    ProjectNames projectNames = GetTestProjectNames(projectFullPaths[i], $"project{i}");
+                    projectCache.AddProjectRestoreInfo(projectNames, projectRestoreInfo, new List<IAssetsLogMessage>());
+                }
+
+                IEnumerable<PackageReference> initialInstalledPackages = (await netCorePackageReferenceProjects[numberOfProjects - 1].GetInstalledPackagesAsync(CancellationToken.None)).ToList();
+
+                // Act
+                IEnumerable<NuGetProjectAction> actions = await nuGetPackageManager.PreviewProjectsUninstallPackageAsync(
+                    netCorePackageReferenceProjects, // All projects
+                    packageB.Id,
+                    uninstallationContext,
+                    testNuGetProjectContext,
+                    CancellationToken.None);
+                using (SourceCacheContext sourceCacheContext = new SourceCacheContext())
+                {
+                    await nuGetPackageManager.ExecuteNuGetProjectActionsAsync(
+                        netCorePackageReferenceProjects,
+                        actions,
+                        testNuGetProjectContext,
+                        sourceCacheContext,
+                        CancellationToken.None);
+                }
+
+                // Assert
+                Assert.Equal(initialInstalledPackages.Count(), 2);
+                List<BuildIntegratedProjectAction> builtIntegratedActions = actions.OfType<BuildIntegratedProjectAction>().ToList();
+                Assert.Equal(actions.Count(), builtIntegratedActions.Count);
+                Assert.True(builtIntegratedActions.All(b => b.RestoreResult.Success));
+                Assert.True(builtIntegratedActions.All(b => !b.RestoreResult.LogMessages.Any())); // There should be no error or warning.
+                List<string> uninstalledLogs = testNuGetProjectContext.Logs.Value.Where(l => l.StartsWith("Successfully uninstalled ")).ToList();
+                Assert.Equal(uninstalledLogs.Count(), 1); // There is bug in Package manager log it only mention 1 project is "Successfully uninstalled " instead of 4.
```

Comment:
Is there a tracking issue for this bug?
Is it an actual bug or is it inconvenient?

Reply:
Sorry, it's more of "inconvenient" part. I just called it bug. I'll change the wording.